bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java

```diff
@@ -0,0 +1,431 @@
+                    String type = tokenizer.poll();
+                    if ("date".equals(type)) {
+                        address = tokenizer.poll() + "/1";
+                        value = tokenizer.poll();
+                    } else if ("time".equals(type)) {
+                        address = tokenizer.poll() + "/0";
+                        value = tokenizer.poll();
+                    } else if (!"request_refresh".equals(type)) {
+                        logger.warn("Received unknown clock event: {}", status);
+                    }
+                    if (value != null && !value.isEmpty()) {
+                        updateGroup(address, value);
+                    }
+                    break;
+                }
+                default:
+                    if ("lighting".equals(commentTokenizer.peek())) {
+                        commentTokenizer.poll();
+                        @Nullable
+                        String commentToken = commentTokenizer.peek();
+                        if ("SyncUpdate".equals(commentToken)) {
+                            commentTokenizer.poll();
+                            @Nullable
+                            String address = commentTokenizer.poll();
+
+                            @Nullable
+                            String level = commentTokenizer.poll();
+                            level.replace("level=", "");
+                            updateGroup(address, level);
+                        }
+                    } else {
+                        logger.warn("Received unparsed event: '{}'", status);
+                    }
+                    break;
+            }
+        }
+
+    }
+
+    @NonNullByDefault
+    private class EventMonitor extends EventCallback {
+
+        @Override
+        public boolean acceptEvent(int event_code) {
+            return true;
+        }
+
+        @SuppressWarnings({ "null" })
+        @Override
+        public void processEvent(@Nullable CGateSession cgate_session, int eventCode,
+                @Nullable GregorianCalendar event_time, @Nullable String event) {
+            if (event == null) {
+                return;
+            }
+            LinkedList<String> tokenizer = new LinkedList<String>(Arrays.asList(event.trim().split("\\s+")));
+
+            if (eventCode == 701) {
+                @Nullable
+                String address = tokenizer.poll();
+                tokenizer.poll();
+                @Nullable
+                String value = tokenizer.poll();
+                if (value != null && value.startsWith("level=")) {
+                    String level = value.replace("level=", "");
+                    updateGroup(address, level);
+                }
+            }
+        }
+
+    }
+
+    private void updateGroup(@Nullable String address, @Nullable String value) {
+        if (address == null || value == null) {
+            return;
+        }
+        String[] addressParts = address.trim().replace("//", "").split("/");
```

Comment:
Unless I'm misunderstanding, you could probably simplify this with regex.
```suggestion
        String[] addressParts = address.trim().split("(//)*/");
```

Reply:
That didnt work. But the trim was unnecessary and i've split the code out to better validate the address string.